### PR TITLE
SDPA-40 adds tide_workflow_notification module

### PR DESCRIPTION
**Pull request by vincent-gao**
# Issues
- I think this submodule should not be enabled as default. 

# Changes
- migrates tide_workflow_notification to tide_core

relates to https://github.com/dpc-sdp/content-vic-gov-au/pull/733

### CODE REVIEW

File: modules/tide_workflow_notification/tide_workflow_notification.module [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        break;
+
+      case 'draft_published':
+        $recipients = _tide_workflow_notification_get_node_approvers($node, $transition);
+        $recipients[$last_author->id()] = $last_author;
+        break;
+
+      // Notify all Approvers and the last author.
+      case 'needs_review_published':
+      case 'published_archived':
```
Comment (anthony-malkoun) [inline modules/tide_workflow_notification/tide_workflow_notification.module]:
```suggestion
      // Notify all Approvers and the last author.
      case 'needs_review_published':
      case 'published_archived':
      case 'draft_published':

```

Could the switch statement just include `draft_published` here?

File: modules/tide_workflow_notification/tide_workflow_notification.module [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+      // Notify all Approvers.
+      case 'draft_needs_review':
+        $recipients = _tide_workflow_notification_get_node_approvers($node, $transition);
+        break;
+
+      case 'draft_published':
+        $recipients = _tide_workflow_notification_get_node_approvers($node, $transition);
+        $recipients[$last_author->id()] = $last_author;
+        break;
```
Comment (anthony-malkoun) [inline modules/tide_workflow_notification/tide_workflow_notification.module]: This does exactly the same as the next block... roll them in together?

File: modules/tide_workflow_notification/tide_workflow_notification.module [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+      // Notify all Approvers and the last author.
```
Comment (anthony-malkoun) [inline modules/tide_workflow_notification/tide_workflow_notification.module]:
```suggestion
      // Notify all Approvers and the last author.
      case 'needs_review_published':
      case 'published_archived':
      case 'draft_published':

```

Could the switch statement just include `draft_published` here?

File: modules/tide_workflow_notification/tide_workflow_notification.module (hunk 2 above)
Comment (vincent-gao) [reply modules/tide_workflow_notification/tide_workflow_notification.module]: Hi @anthony-malkoun , it is fixed.

File: modules/tide_workflow_notification/tide_workflow_notification.module (hunk 1 above)
Comment (vincent-gao) [reply modules/tide_workflow_notification/tide_workflow_notification.module]: Hi @anthony-malkoun , fixed. thanks

**Pull request merged**